### ISSUE TITLE
editorconfig 0.11.x uninstallable since npm v2.7.6

### ISSUE THREAD

**Issue by quarterto** [npm/npm#7992]
```
❯ npm install editorconfig@0.11
npm WARN package.json editorconfig@0.11.4 No bin file found at bin\editorconfig
npm ERR! Darwin 14.1.0
npm ERR! argv "node" "/usr/local/bin/npm" "install" "editorconfig@0.11"
npm ERR! node v0.10.36
npm ERR! npm  v2.8.3
npm ERR! path /Users/matthew.brennan/Documents/npm/node_modules/editorconfig/bin\editorconfig
npm ERR! code ENOENT
npm ERR! errno 34

npm ERR! enoent ENOENT, chmod '/Users/matthew.brennan/Documents/npm/node_modules/editorconfig/bin\editorconfig'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
```

I've bisected npm, running (effectively) `git bisect run bin/npm-cli.js install editorconfig@0.11`, which points to da015ee. I suspect (but have no idea how) ripping out the caching logic introduced this bug.

Other things I don't know:
1. Where the `\` comes from in the bin path
2. Why `editorconfig@0.12` installs just fine when the relevant bit of `package.json` [hasn't changed](https://github.com/editorconfig/editorconfig-core-js/compare/v0.11.4...v0.12.2#diff-6)

Here's my [npm-debug.log](https://gist.github.com/anonymous/b6f6282002d22fc0a408), just in case. For reference, the original bug I'm tracking down was Financial-Times/origami-build-tools#245.

(paging @othiym23)

**Comment by smikes** [comment]
Here's part of the puzzle.  The `\` is present in the online version of package.json for editorconfig@0.11 --

```
$ npm view editorconfig@0.11.4 | grep bin
  directories: { bin: './bin', lib: './lib' },
  bin: { editorconfig: 'bin\\editorconfig' },
```

Note that this is **not** present in the `package/package.json` that is included in `editorconfig-0.11.4.tgz`

I suspect a race condition.  One time, I was able to install editorconfig@0.11 by running

```
npm -ddd install editorconfig@0.11
```

But that only worked once -- more often, I saw the failure documented in OP.  

I think this is one for the general category of `package.json` differs between package and registry, and there are race conditions surrounding which one is used/read.
